### PR TITLE
Fix ``sym is no accessible`` while iterating record (#5354)

### LINKED ISSUE #5354: ``sym is not accessible`` in compile time
```nim
type
  Test = object
    case p: bool
    of true:
      a: int
    else:
      discard

proc f[T](t: typedesc[T]): int =
  1

when isMainModule:
  var x = Test(p: true, a: 1)
  doAssert Test.f == 1
```

produces:
```
Traceback (most recent call last)
nim.nim(121)             nim
nim.nim(77)              handleCmdLine
main.nim(163)            mainCommand
main.nim(74)             commandCompileToC
modules.nim(240)         compileProject
modules.nim(180)         compileModule
passes.nim(213)          processModule
passes.nim(135)          processTopLevelStmt
sem.nim(495)             myProcess
sem.nim(467)             semStmtAndGenerateGenerics
semstmts.nim(1609)       semStmt
semexprs.nim(819)        semExprNoType
semexprs.nim(2283)       semExpr
semexprs.nim(1847)       semWhen
semexprs.nim(2345)       semExpr
semstmts.nim(1563)       semStmtList
semexprs.nim(2252)       semExpr
semexprs.nim(802)        semDirectOp
semexprs.nim(640)        semOverloadedCallAnalyseEffects
semcall.nim(384)         semOverloadedCall
semcall.nim(199)         resolveOverloads
semcall.nim(87)          pickBestCandidate
sigmatch.nim(1789)       matches
sigmatch.nim(1746)       matchesAux
sigmatch.nim(1583)       prepareOperand
semexprs.nim(26)         semOperand
semexprs.nim(2267)       semExpr
semexprs.nim(1839)       semMagic
semexprs.nim(802)        semDirectOp
semexprs.nim(640)        semOverloadedCallAnalyseEffects
semcall.nim(384)         semOverloadedCall
semcall.nim(199)         resolveOverloads
semcall.nim(87)          pickBestCandidate
sigmatch.nim(1789)       matches
sigmatch.nim(1746)       matchesAux
sigmatch.nim(1583)       prepareOperand
semexprs.nim(26)         semOperand
semexprs.nim(2225)       semExpr
semexprs.nim(1167)       semFieldAccess
semexprs.nim(1100)       builtinFieldAccess
system.nim(2590)         sysFatal
Error: unhandled exception: sym is not accessible [FieldError]
```

### PR DESCRIPTION
**Pull request by vegansk**
Fixes #5354

### CODE REVIEW

File: compiler/semexprs.nim [hunk 1]
```diff
@@ -1094,7 +1094,7 @@ proc builtinFieldAccess(c: PContext, n: PNode, flags: TExprFlags): PNode =
     of tyObject, tyTuple:
       if ty.n != nil and ty.n.kind == nkRecList:
         for field in ty.n:
-          if field.sym.name == i:
+          if field.kind == nkSym and field.sym.name == i:
```
Comment (Araq) [inline compiler/semexprs.nim]: I will accept this, but the original code is all wrong. It needs to iterate over the 'ty.n' properly.

Comment (vegansk) [inline compiler/semexprs.nim]: Did I get it wrong? It iterates other the 'ty.n' in line 1096. Maybe it needs to iterate other all the fields of the object variants here?

Comment (Araq) [inline compiler/semexprs.nim]: I think so, yes.

**Comment by Araq** [comment]
Fixed it differently.

**Pull request closed without merging**